packages/babel-helpers/src/helpers/regenerator.ts

```diff
@@ -166,389 +171,223 @@ export default function /* @no-mangle */ _regenerator() {
   function makeInvokeMethod(
     innerFn: Function,
     self: unknown,
-    context: Context,
+    tryLocsList: TryLocs[],
   ) {
     var state = GenState.SuspendedStart;
 
-    return function invoke(method: "next" | "throw" | "return", arg: any) {
+    function invoke(
+      _methodName: "next" | "throw" | "return",
+      _method: OperatorType.Next | OperatorType.Throw | OperatorType.Return,
+      _arg: any,
+    ) {
       if (state === GenState.Executing) {
-        throw new Error("Generator is already running");
-      }
-
-      if (state === GenState.Completed) {
-        if (method === "throw") {
-          throw arg;
+        throw Error("Generator is already running");
+      } else if (done) {
+        if (_method === OperatorType.Throw) {
+          Context_dispatchExceptionOrFinishOrAbrupt(_method, _arg);
         }
-
-        // Be forgiving, per GeneratorResume behavior specified since ES2015:
-        // ES2015 spec, step 3: https://262.ecma-international.org/6.0/#sec-generatorresume
-        // Latest spec, step 2: https://tc39.es/ecma262/#sec-generatorresume
-        return { value: undefined, done: true };
       }
 
-      context.method = method;
-      context.arg = arg;
-
-      while (true) {
-        var delegate = context.delegate;
-        if (delegate) {
-          var delegateResult = maybeInvokeDelegate(delegate, context);
-          if (delegateResult) {
-            if (delegateResult === ContinueSentinel) continue;
-            return delegateResult;
+      method = _method;
+      arg = _arg;
+
+      while (!done || (_ = undefined)) {
+        if (!delegateIterator) {
+          if (!method /* Return */) {
+            ctx.sent = arg;
+          } else if (method < 3 /* Throw | Return */) {
+            if (method > 1 /* Return */) ctx.next = ContextNext.End;
+            Context_dispatchExceptionOrFinishOrAbrupt(method, arg);
+          } else {
+            // Jump
+            ctx.next = arg;
           }
         }
-
-        if (context.method === "next") {
-          // Setting context._sent for legacy support of Babel's
-          // function.sent implementation.
-          context.sent = context._sent = context.arg;
-        } else if (context.method === "throw") {
-          if (state === GenState.SuspendedStart) {
-            state = GenState.Completed;
-            throw context.arg;
+        try {
+          if (delegateIterator) {
+            // Call delegate.iterator[context.method](context.arg) and handle the result
+
+            if (
+              (_ = delegateIterator[FunctionNameStrings[method as 0 | 1 | 2]])
```

Comment:
We can use `_methodName` here.

Reply:
Oh I see, we cannot because `Context_dispatchExceptionOrFinishOrAbrupt` might change `method`.

Reply:
I tried that and it results in an error, I'll look into it again. :)